include/ts/apidefs.h.in

```diff
@@ -985,6 +990,23 @@ using TSMgmtInt     = int64_t;
 using TSMgmtCounter = int64_t;
 using TSMgmtFloat   = float;
 using TSMgmtString  = char *;
+using TSMgmtSeconds = std::chrono::seconds;
+
+class HttpStatusCodeList;
+struct HostResData;
+class HttpForwardedConf;
+
+// clang-format off
+using TSConfigValue = std::variant<
+                        TSMgmtInt,
+                        TSMgmtFloat,
+                        TSMgmtSeconds,
+                        std::string_view,
+                        HttpStatusCodeList *,
+                        HostResData *,
+                        HttpForwardedConf *,
+                        TSOutboundConnectionMatchType>;
```

Comment:
Now, this variant captures all types in the `OverridableHttpConfigParams`. However, as @zwoop pointed out, every time we add a new type here, ABI compatibility is broken 🤔